elftools/elf/sections.py

```diff
@@ -500,8 +467,133 @@ def _make_subsections(self):
+                    s_number = struct_parse(structs.Elf_uleb128('s_number'),
+                                            stream
+                               )
+
+        elif self.tag in ('TAG_CPU_RAW_NAME', 'TAG_CPU_NAME', 'TAG_CONFORMANCE'):
+            self.value = struct_parse(structs.Elf_ntbs('value',
+                                                       encoding='utf-8'),
+                                      stream)
+
+        elif self.tag == 'TAG_COMPATIBILITY':
+            self.value = struct_parse(structs.Elf_uleb128('value'), stream)
+            self.extra = struct_parse(structs.Elf_ntbs('vendor_name',
+                                                       encoding='utf-8'),
+                                      stream)
+
+        elif self.tag == 'TAG_ALSO_COMPATIBLE_WITH':
+            self.value = ARMAttribute(structs, stream)
+
+            if type(self.value.value) is not str:
+                nul = struct_parse(structs.Elf_byte('nul'), stream)
+                elf_assert(nul == 0,
+                           "Invalid terminating byte %r, expecting NUL." % nul)
+
+        else:
+            self.value = struct_parse(structs.Elf_uleb128('value'), stream)
+
+
+class ARMAttributesSubsubsection(AttributesSubsubsection):
+    """ Subsubsection of an ELF .ARM.attributes section's subsection.
+    """
+    def __init__(self, stream, structs, offset):
+        super(ARMAttributesSubsubsection, self).__init__(
+            stream, structs, offset, ARMAttribute)
+
+
+class ARMAttributesSubsection(AttributesSubsection):
+    """ Subsection of an ELF .ARM.attributes section.
+    """
+    def __init__(self, stream, structs, offset):
+        super(ARMAttributesSubsection, self).__init__(
+            stream, structs, offset,
+            structs.Elf_Attr_Subsection_Header,
+            ARMAttributesSubsubsection
+        )
+
+
+class ARMAttributesSection(AttributesSection):
+    """ ELF .ARM.attributes section.
+    """
+    def __init__(self, header, name, elffile):
+        super(ARMAttributesSection, self).__init__(
+            header, name, elffile, ARMAttributesSubsection)
+
+
+class RISCVAttribute(Attribute):
+    """ Attribute of an ELF .riscv.attributes section.
+    """
+    def __init__(self, structs, stream):
+        super(RISCVAttribute, self).__init__(
+            struct_parse(structs.Elf_RiscV_Attribute_Tag, stream))
+
+        if self.tag in ('TAG_FILE', 'TAG_SECTION', 'TAG_SYMBOL'):
+            self.value = struct_parse(structs.Elf_word('value'), stream)
+
+            if self.tag != 'TAG_FILE':
+                self.extra = []
+                s_number = struct_parse(structs.Elf_uleb128('s_number'), stream)
+
+                while s_number != 0:
+                    self.extra.append(s_number)
+                    s_number = struct_parse(structs.Elf_uleb128('s_number'),
+                                            stream
+                                            )
```

Comment:
here too -- `)` up

Reply:
And done. I also changed two other occurrences that were already there.